### PR TITLE
DCOS-44049: Add AppChrome components

### PR DESCRIPTION
**Pull request by mperrotti**
Adds components to allow Mesosphere engineers to quickly create a chrome for their app that matches Mesosphere's design patterns

**Note to reviewers:**
I highly recommend taking this commit-by-commit. The most important stuff is in the last commit.

appchrome

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -26,7 +26,8 @@ module.exports = {
       "setupTestFrameworkScriptFile": "<rootDir>/testHelper/setupTests.ts",
       "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
       "transform": {
-        "^.+\\.tsx?$": "ts-jest"
+        "^.+\\.tsx?$": "ts-jest",
+        "^.+\\.js?$": "ts-jest"
```
Comment (mperrotti) [inline jest.config.js]: Needed this for tests to run on components that have imports from `design-tokens`. Is there a better solution?

**Comment by mperrotti** [comment]
Hold merge until Lily gives me feedback

File: jest.config.js (hunk 1 above)
Comment (mperrotti) [reply jest.config.js]: I ended up changing the design-tokens JS exports to commonJS, but it still appears to be failing in CI :(

File: packages/appChrome/components/SidebarSubMenuItem.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  SidebarSubMenuItemProps,
+  {}
+> {
+  public render() {
+    const { children, isActive, onClick } = this.props;
+    const classNames = cx(
+      subMenuItem,
+      appChromeInsetContent,
+      sidebarNavItem(isActive),
+      // TODO: ask Lily about using our secondary text color/s instead
```
Comment (GeorgiSTodorov) [inline packages/appChrome/components/SidebarSubMenuItem.tsx]: Is fixing this in the scope of this ticket or just a reminder for the future?

File: packages/expandable/style.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { css } from "emotion";
+
+// TODO: better name than "details"?
```
Comment (GeorgiSTodorov) [inline packages/expandable/style.ts]: Should this be here?

File: packages/appChrome/components/SidebarSubMenuItem.tsx (hunk 2 above)
Comment (mperrotti) [reply packages/appChrome/components/SidebarSubMenuItem.tsx]: Lily and I have a meeting to review some of these design-related "TODO"s at 10:30(PST)

File: packages/appChrome/tests/Sidebar.test.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+          Item label
+        </SidebarItemLabel>
+      );
+      expect(toJson(component)).toMatchSnapshot();
+    });
+  });
+
+  describe("SidebarSubMenu", () => {
+    const onClickFn = jest.fn();
+    // tslint:disable:jsx-wrap-multiline
```
[the remark this answers is not in the document]
Comment (mperrotti) [reply packages/appChrome/tests/Sidebar.test.tsx]: Yes. It compiles just fine without it, I just don't like seeing the red underline in my text editor 😬

File: packages/appChrome/components/SidebarSubMenuItem.tsx (hunk 2 above)
Comment (GeorgiSTodorov) [reply packages/appChrome/components/SidebarSubMenuItem.tsx]: After the meeting will you update this PR or make a new one?

File: packages/appChrome/style.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+export const sidebar = css`
+  background-color: ${greyDark};
+  width: ${sidebarWidths.default};
+
+  ${atMediaUp[layoutBreakpoint](css`
+    width: ${sidebarWidths[layoutBreakpoint]};
+  `)};
+`;
+
+// TODO: replace animation duration/easing with design tokens
```
[the remark this answers is not in the document]
Comment (mperrotti) [reply packages/appChrome/style.ts]: This one is probably future consideration. I don't think Lily and I will be able to finalize our plans for animation-related design tokens in our design review meeting

File: packages/appChrome/components/SidebarSubMenuItem.tsx (hunk 2 above)
Comment (mperrotti) [reply packages/appChrome/components/SidebarSubMenuItem.tsx]: Update this one

File: packages/appChrome/tests/Sidebar.test.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    const subMenuItems = [
+      <SidebarSubMenuItem key={0} onClick={onClickFn}>
+        Item label
+      </SidebarSubMenuItem>,
+      <SidebarSubMenuItem key={1} onClick={onClickFn}>
+        Item label
+      </SidebarSubMenuItem>
+    ];
+    // tslint:enable
```
[the remark this answers is not in the document]
Comment (mperrotti) [reply packages/appChrome/tests/Sidebar.test.tsx]: Yes, that turns ts-lint back on. I just didn't want to lint those two JSX elements

File: packages/expandable/style.ts (hunk 3 above)
Comment (mperrotti) [reply packages/expandable/style.ts]: No :x

File: packages/appChrome/components/SidebarSubMenuItem.tsx (hunk 2 above)
Comment (GeorgiSTodorov) [reply packages/appChrome/components/SidebarSubMenuItem.tsx]: So it's still work in progress.

Comment (mperrotti) [reply packages/appChrome/components/SidebarSubMenuItem.tsx]: Yes, but the only things left to do should be minor visual design tweaks. If there are any changes that take more than a few minutes to add, they'll go into a new ticket

Comment (GeorgiSTodorov) [reply packages/appChrome/components/SidebarSubMenuItem.tsx]: Okay, cool. Still, adding new things dismisses old reviews which is why I asked.

**Pull request merged**